homeassistant/components/soma/__init__.py

```diff
@@ -99,3 +106,38 @@ def device_info(self) -> DeviceInfo:
             manufacturer="Wazombi Labs",
             name=self.name,
         )
+
+    def set_position(self, position: int) -> None:
+        """Set the current device position."""
+        self.current_position = position
+        self.schedule_update_ha_state()
+
+    async def async_get_state_from_api(self) -> dict:
```

Comment:
This is substantially similar to `async_update` in `sensor.py`

You could make this a decorator to wrap around the api call

Reply:
Nice idea! I have had trouble with it but I think I got it right now. Tested and functional.
Let me know what you think!